examples/events.rs

```diff
@@ -0,0 +1,134 @@
+//! Adapted from https://github.com/bevyengine/bevy/blob/v0.9.1/examples/2d/sprite_sheet.rs
+//! and https://github.com/bevyengine/bevy/blob/v0.9.1/examples/animation/animated_fox.rs
+//! Renders an animated sprite by loading all animation frames from multiple sprites
+//! and changing the displayed image periodically.
+
+#[path = "helpers/animation_controller.rs"]
+mod animation_helper;
+
+use animation_helper::keyboard_animation_control_helper;
+use bevy::prelude::*;
+use bevy_trickfilm::prelude::*;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest())) // prevents blurry sprites
+        .add_plugins(Animation2DPlugin)
+        .add_event::<SampleEvent>()
+        .add_animation_event::<SampleEvent>()
+        .register_type::<SampleEvent>()
+        .add_systems(Startup, setup)
+        .add_systems(
+            Update,
+            (keyboard_animation_control, update_frame_text, print_event),
+        )
+        .run();
+}
+
+#[derive(Debug, Clone, Event, Reflect)]
+struct SampleEvent {
+    #[reflect(skip_serializing)]
+    #[reflect(default = "default_entity")]
+    entity: Entity,
```

Comment:
Possible alternative would be to use `Option<Entity>` which make quite a few things easier I reckon. So I am sure you considered this option as well, what was the reason for going with default entity?

Reply:
I did not consider that option. You are right that it makes things a bit simpler here (I really do not want to provide that default_entity function :D). The downside is that when you react on the Event and want to retrieve the entity you need to unwrap.
I will probably play around with macros to derive AnimationEvent and fix it that way.

Reply:
Oh haha, using Option is always my first instinct. But yeah the user would need to unwrap the thing which is a bit annoying, especially because the entity should always be valid if the animation event triggered (though couldn't it be that with a frame delay the entity gets despawned in the same frame as it triggers an animation event? Then if you listen for the animation event one frame later the entity will already be destroyed but you are still trying to reference it, though even with an Option that would still be a problem).

Reply:
Yeah, Entity should always be valid as far as the event is concerned. I know provide a thin wrapper around Entity that derefs to Entity.

About despawned entities: Entity is just an id. You always need to interact with the archetype table. I would expect that specific id to be invalidated on that side. Looks like commands.entity panics and commands.get_entity returns None.

Reply:
Oh yeah true, I didn't think about that, the user will need to handle that anyways. Though I don't quite understand why you use a wrapper, does it make the implementation easier or something? As the user I don't quite understand how it improves the API (as it really is just a wrapper at this point).

Reply:
FromReflect (or Reflect?) needs all struct members to implement Default, which Entity does not.
That is why you need to annotate the member with reflect(default). By using a Wrapper, just the wrapper can impl Default and you don't need to worry about it.
Not perfect, but better than nothing

Reply:
Ahh okay yeah that makes perfect sense.